alf/networks/networks.py

```diff
@@ -320,10 +320,15 @@ def forward(self, input, state):
 @alf.configurable
 @alf.repr_wrapper
 class NoisyFC(Network):
-    """The Noisy Linear Layer discribed in
+    r"""The Noisy Linear Layer discribed in
 
     Fortunato et. al. `Noisy Networks for Exploration <https://arxiv.org/abs/1706.10295>`_
 
+    In short, the original weight :math:`w` and bias :math:`b` of FC layer are replaced
+    with :math:`w + w_\sigma \epislon^w` and :math:`b + b_\sigma \epsion^b` where
+    :math:`\epsilon^w` and :math:`\epsilon^b` are noise and :math:`w, w_\sigma, b, b_\sigma`
```

Comment:
This comment of what the noise is applied to is inconsistent with the actual code. The comment says the noise is multiplied to the weight while the code multiplies it with the inputs. For the former, the noise of each sample in a batch is the same.

Reply:
The noise is different for each sample. And the noise is factorized \epsion_w = \epsion_in \otimes \epsilon_out

Reply:
> The noise is different for each sample. And the noise is factorized \epsion_w = \epsion_in \otimes \epsilon_out

Yeah I know the code has different noises for different samples. But just from the code here, ":math:`w + w_\sigma \odot \epislon^w`" (the noises are directly applied to the single weight matrix), the noise is the same for different samples?

Reply:
It is equivalent to adding the noise to the weight matrix. And for each sample, the noise is different.

Reply:
> It is equivalent to adding the noise to the weight matrix. And for each sample, the noise is different.

I can get the equivalence. I was just saying that it's easy for someone to misunderstand the noise as the same for different inputs, if not combining the comment here with the details 1 below.